### PR TITLE
Rust: Move to bzlmod.

### PR DESCRIPTION
**Pull request by criemen**
This gets rid of our last workspace dependency.
In particular, this change also gets rid of the checked-in extra
lock files that took forever to generate.

This PR also upgrades bazel to 7.2.0 while we're at it, and upgrades a bunch of external dependencies.

### CODE REVIEW

**Review by tausbn — APPROVED** [review]
Hooray! 🎉 
Build changes look sensible to me (but the Bazel upgrade seems like it might have broken something?).

**Review by aibaars — APPROVED** [review]

**Comment by criemen** [comment]
We need an internal PR for this, as the path to the build targets change, otherwise I'm not aware that anything is broken.
But I see the backlink to this PR is missing, let me fix that.

**Review by erik-krogh — APPROVED** [review]

File: misc/bazel/rust.bzl [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+load("@semmle_code//buildutils-internal:glibc_symbols_check.bzl", "glibc_symbols_check")
+load("@semmle_code//buildutils-internal:lipo.bzl", "universal_binary")
+
+def codeql_rust_binary(
+        name,
+        target_compatible_with = None,
+        visibility = None,
+        symbols_test = True,
+        **kwargs):
+    rust_label_name = name + "_single_arch"
```
Comment (redsun82) [inline misc/bazel/rust.bzl:11]: wouldn't it be maybe better to use `single_arch/name`, to get a consistent binary base name?

File: misc/bazel/rust.bzl [hunk 2]
```diff
@@ -0,0 +1,23 @@
+load("@rules_rust//rust:defs.bzl", "rust_binary")
+load("@semmle_code//buildutils-internal:glibc_symbols_check.bzl", "glibc_symbols_check")
+load("@semmle_code//buildutils-internal:lipo.bzl", "universal_binary")
```
Comment (redsun82) [inline misc/bazel/rust.bzl:3]: not in the context of this PR, but we probably should think about moving this rule in this repo or go with the separate bazel lib repo idea we had

**Review by redsun82 — APPROVED** [review]
Thanks for this! Looks good on the bazel side, with just a couple of comments that can be freely ignored

**Pull request merged**